peer.go

```diff
@@ -306,11 +307,13 @@ func (p *peer) loadActiveChannels(chans []*channeldb.OpenChannel) error {
 		// Register this new channel link with the HTLC Switch. This is
 		// necessary to properly route multi-hop payments, and forward
 		// new payments triggered by RPC clients.
-		sphinxDecoder := htlcswitch.NewSphinxDecoder(p.server.sphinx)
 		link := htlcswitch.NewChannelLink(
 			htlcswitch.ChannelLinkConfig{
-				Peer:             p,
-				DecodeOnion:      sphinxDecoder.Decode,
+				Peer:                  p,
+				DecodeHopIterator:     p.server.sphinx.DecodeHopIterator,
+				DecodeOnionObfuscator: p.server.sphinx.DecodeOnionObfuscator,
+				GetLastChannelUpdate: createGetLastUpdate(p.server.chanRouter,
+					p.PubKey(), chanID),
```

Comment:
We have access to the channel point here above, so we can pass that in directly. If we do this, this let's us avoid looping over all the edges in the entire graph each time we need to send an error which requires an update.